app/src/main/java/com/alphawallet/app/service/KeyService.java

```diff
@@ -616,10 +616,10 @@ private synchronized boolean storeEncryptedBytes(byte[] data, boolean createAuth
             boolean success = writeBytesToFile(ivPath, iv);
             if (!success)
             {
-                deleteKey(fileName);
+                //deleteKey(fileName);
```

Comment:
@JamesSmartCell better to simply delete the code rather than keep it round

Reply:
There's a reason why I did this - 

with biometrics on Samsung phone certain biometric unlock doesn't result in a 'secure unlock' event, only a 'succeeded in proving face biometric' so the storeEncryptedBytes would fail if upgrading.

In this case we definitely want the key to remain otherwise the user has lost their key.

Since we can't completely rule out this issue in the future it's best to err on the side of caution and not delete keys if there's an issue with storing the key in the case where we're upgrading the key.